environment.cpp

```diff
@@ -103,17 +103,19 @@ std::vector<double> create_n_inputs(std::uniform_real_distribution<double> dist,
 }
 
 
-void switch_env_function(environment &e)
+environment switch_env_function(environment e)
```

Comment:
This could stay avoid, but nice to conver it to a return type. ususally this optimized away, I am not sure in this case.

Reply:
I am saying this because this is not a read-only / getter function

Reply:
How could this be a void without actually passing something by reference? It's just going to do some actions to a copy of something and then throw the copy away, no?

Reply:
It cannot, you are right. But what I am saying is that getters should take const reference, but free functions can take by reference and do things to the object. They just have to use member functions of the object that are not getters.
In this case they should use `environment::chang_env_function()` or something like that.
In general anyway I think it could be a good idea to give free functions a return, the only thing is that we will get problems with `calc_fitness(population)` and `reproduce(simulation)`